pkg/auditlog/auditlog.go

```diff
@@ -63,12 +61,13 @@ type Event struct {
 	ActorID    string     `json:"actor_id"`
 	Group      string     `json:"group"`
 
-	// A key that ensures that an same event is not processed multiple time.
+	// A key that ensures that the same event is not processed multiple times.
 	// Once the event is sent for the first time, the lock on the key expires
 	// after 24 hours.
-	//
-	// An idempotency key is automatically generated if not set.
-	IdempotencyKey string `json:"-"`
+
+	// If no key is provided or the key is empty, the key will not be attached
+	// to the request.
+	IdempotencyKey string
```

Comment:
This will include the idempotency key in the JSON payload under the `IdempotencyKey`.
Since we are including it in the header I don't think we should remove the `json:"-"`